OpenRA.Mods.Common/Graphics/ContrailRenderable.cs

```diff
@@ -66,11 +67,24 @@ public void Render(WorldRenderer wr)
 			// Start of the first line segment is the tail of the list - don't smooth it.
 			var curPos = trail[Index(next - skip - 1)];
 			var curColor = color;
-			for (var i = 0; i < length - skip - 4; i++)
+
+			for (var i = 1; i < renderLength; i++)
 			{
-				var j = next - skip - i - 2;
-				var nextPos = Average(trail[Index(j)], trail[Index(j - 1)], trail[Index(j - 2)], trail[Index(j - 3)]);
-				var nextColor = Exts.ColorLerp(i * 1f / (length - 4), color, Color.Transparent);
+				var j = next - skip - 1 - i;
+				WPos nextPos;
+
+				// Smooth the contrail to tail direction, use 4 contrail position for max smooth effect.
```

Comment:
This comment doesn't really help, so I suggest either rewording or removing it.

Reply:
The problem here is that "max smooth effect" is meaningless. The comments here should be explaining about the circular buffer, which is a critical detail in this implementation that I had not picked up on!

Reply:
ok, ok. Because the circular buffer related did not appear in comments so I ignore it. Should not the person do this function gave comments here? and I think circular buffer should belong to somewhere else in this module.

Reply:
line 26
		// Store trail positions in a circular buffer
		readonly WPos[] trail;